src/EurekaClient.js

```diff
@@ -51,6 +51,10 @@ export default class Eureka extends EventEmitter {
 
     this.logger.debug('initializing eureka client');
 
+    if (typeof config.requestMiddleware === 'function') {
```

Comment:
Thanks for the suggestion, I can add a noop to the defaultConfig which will cause it to default to a function instead of `undefined`, but I would like to keep the type checking in there for some degree of type safety.